### PR TITLE
Add verbose option.

### PR DESCRIPTION
**Pull request by Daniel-Dodd**
Simple implementation of verbosity option #115 . 

Tests don't check if this works or not, but only if any error occurs when turning verbosity on/off.

Training loops tests have been updated to ensure testing of edge case of a single iteration.

### CODE REVIEW

File: gpjax/abstractions.py [hunk 1]
```diff
@@ -175,6 +178,7 @@ def fit_batches(
     batch_size: int,
     n_iters: Optional[int] = 100,
     log_rate: Optional[int] = 10,
+    verbose: Optional[bool] = True,
```
Comment (thomaspinder) [inline gpjax/abstractions.py:181]: Do you think defaulting to True is sensible? I can see benefits for defaulting to False or True, but the former feels less invasive? Keen to get your thoughts though.

**Review by thomaspinder — COMMENTED** [review]
My only comment is regarding the verbosity argument's default value. I have no strong feelings but I am curious to hear your thoughts.
